.github/workflows/i18n.yml

```diff
@@ -0,0 +1,135 @@
+---
+name: I18n
+on:
+  schedule:
+    - cron: '0 7 * * 0,4'
+
+env:
+  JAVA_DISTRIBUTION: 'temurin'
+
+jobs:
+  main_branch_i18n:
+    name: main branch i18n
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:38
+
+    steps:
+      - name: Mask secrets
+        shell: bash
+        run: |
+          echo "::add-mask::${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Install dependencies
+        shell: bash
+        run: dnf --setopt install_weak_deps=False install -y gettext jss git
+
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: 17
+
+      - name: Add safe directory
+        shell: bash
+        run: git config --system --add safe.directory $GITHUB_WORKSPACE
+
+      - name: Run internationlization
+        shell: bash
+        run: |
+          chmod +x ./.github/scripts/i18n.sh
+          ./.github/scripts/i18n.sh
+        env:
+          GIT_AUTHOR_EMAIL: '3229038+candlepin-bot@users.noreply.github.com'
+          GIT_AUTHOR_NAME: 'candlepin-bot'
+          GIT_BRANCH: 'main'
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  4.1_branch_i18n:
+    name: 4.1 branch i18n
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:38
+
+    steps:
+      - name: Mask secrets
+        shell: bash
+        run: |
+          echo "::add-mask::${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Install dependencies
+        shell: bash
+        run: dnf --setopt install_weak_deps=False install -y gettext jss git
+
+      - name: Check out repository
+        uses: actions/checkout@v3
+        with:
+          ref: candlepin-4.1-HOTFIX
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: 11
+
+      - name: Add safe directory
+        shell: bash
+        run: git config --system --add safe.directory $GITHUB_WORKSPACE
+
+      - name: Run internationlization
+        shell: bash
+        run: |
+          chmod +x ./.github/scripts/i18n.sh
+          ./.github/scripts/i18n.sh
+        env:
+          GIT_AUTHOR_EMAIL: '3229038+candlepin-bot@users.noreply.github.com'
+          GIT_AUTHOR_NAME: 'candlepin-bot'
+          GIT_BRANCH: 'candlepin-4.1-HOTFIX'
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  4.2_branch_i18n:
```

Comment:
Just an idea. If we really wanted, we could have this as a reusable workflow in each branch. All that the main scheduled workflow would need to do is to call them from all supported branches.

Something like this.

```yaml
uses: {owner}/{repo}/.github/workflows/i18n.yml@{branch}
```

Reply:
I feel like this makes things harder than they need to be for not much gain as the code stands today. I could see this being pretty useful if a code line branch needs to handle i18n in a drastically different way. Then you could let that logic live in the code line branch and all the scheduled main branch would have to do is call it like you pointed out.